### ISSUE TITLE
[colab] runtime error: no cuda gpus are available :  for e in range(num_epochs) 

### ISSUE THREAD

**Issue by mina-dong** [mina-dong/Univ_NLP#10]
for e in range(num_epochs):
    train_acc = 0.0
    test_acc = 0.0
    model.train()
    for batch_id, (token_ids, valid_length, segment_ids, label) in enumerate(tqdm_notebook(train_dataloader)):
        optimizer.zero_grad()
        token_ids = token_ids.long().to(device)
        segment_ids = segment_ids.long().to(device)
        valid_length= valid_length
        label = label.long().to(device)
        out = model(token_ids, valid_length, segment_ids)
        loss = loss_fn(out, label)
        loss.backward()
        torch.nn.utils.clip_grad_norm_(model.parameters(), max_grad_norm)
        optimizer.step()
        scheduler.step()  # Update learning rate schedule
        train_acc += calc_accuracy(out, label)
        if batch_id % log_interval == 0:
            print("epoch {} batch id {} loss {} train acc {}".format(e+1, batch_id+1, loss.data.cpu().numpy(), train_acc / (batch_id+1)))
    print("epoch {} train acc {}".format(e+1, train_acc / (batch_id+1)))

    model.eval()
    for batch_id, (token_ids, valid_length, segment_ids, label) in enumerate(tqdm_notebook(test_dataloader)):
        token_ids = token_ids.long().to(device)
        segment_ids = segment_ids.long().to(device)
        valid_length= valid_length
        label = label.long().to(device)
        out = model(token_ids, valid_length, segment_ids)
        test_acc += calc_accuracy(out, label)
    print("epoch {} test acc {}".format(e+1, test_acc / (batch_id+1)))

**Comment by mina-dong** [comment]
for e in range(num_epochs): epoch 5설정 = 5번반복

model.train(): 모델을 학습 모드로 설정.

for batch_id, (token_ids, valid_length, segment_ids, label) in enumerate(tqdm_notebook(train_dataloader)): 
학습 데이터셋 호출, 각 배치에 대한 학습진행.

optimizer.zero_grad(): 
옵티마이저의 그래디언트를 초기화. 데이터를 GPU로 전송. 
모델에 데이터를 입력으로 전달하여 출력을 얻고 손실계산. 
역전파를 수행하고 그래디언트 클리핑을 적용. 
옵티마이저를 통해 파라미터를 업데이트 스케줄러를 통해 학습률을 업데이트. 
정확도 계산. 현재의 학습 상태를 출력.( 일정 간격마다) 

현재 코드에서 accuracy 설정되어 있음.

model.eval(): 모델을 평가 모드로 변경.

**Comment by mina-dong** [comment]
runtime error: no cuda gpus are available

gpu 사용 시간이 종료되서 오류 발생... cuda를 통해 gpu로 변경되었는지 확인. 일부 코드 중에 gpu 상태에서만 돌아가게 되어있는게 있음.

#GPU 설정
device = torch.device("cuda:0")
